bokeh/embed/util.py

```diff
@@ -212,37 +288,36 @@ def standalone_docs_json_and_render_items(models):
         log.warn(msg)
 
 
-    render_items = []
-    docs_by_id = {}
-    for p in models:
-        modelid = None
-        if isinstance(p, Document):
-            doc = p
+    docs = {}
+    for model_or_doc in models:
+        if isinstance(model_or_doc, Document):
+            model = None
+            doc = model_or_doc
         else:
-            if p.document is None:
-                raise ValueError("To render a Model as HTML it must be part of a Document")
-            doc = p.document
-            modelid = p._id
-        docid = None
-        for key in docs_by_id:
-            if docs_by_id[key] == doc:
-                docid = key
-        if docid is None:
-            docid = make_id()
-            docs_by_id[docid] = doc
-
-        elementid = make_id()
```

Comment:
presumably this is the reason why the `RenderItem` does not have an `elementid`. Is this intentional?

Reply:
Yes. By default, `render_item.roots` tells where to put particular roots. `render_item.elementid` is meant for the rare situation where pre-mapping roots wasn't possible, e.g. when adding roots after creation and initial rendering of a document. Thus it should be used only when really necessary, e.g. with the server.

Reply:
This was fixed by a `node make clean` see below